src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java

```diff
@@ -120,53 +138,66 @@ public void scenarios_should_behave_as_specified() throws IOException {
 
-            final long testCaseId = MapValueExtractor.extractOrDefault(
-                scenario,
-                "request.input.id",
-                ThreadLocalRandom.current().nextInt(1, 9999999 + 1)
-
+            final long scenarioTestCaseId = MapValueExtractor.extractOrDefault(
+                    scenario,
+                    "request.input.id",
+                    -1
             );
 
+            final long testCaseId = (scenarioTestCaseId == -1)
+                    ? ThreadLocalRandom.current().nextLong(1111111111111111L, 1999999999999999L)
+                    : scenarioTestCaseId;
+
             final String requestBody = buildCallbackBody(
-                testCaseId,
-                MapValueExtractor.extract(scenario, "request.input"),
-                templatesByFilename
+                    testCaseId,
+                    MapValueExtractor.extract(scenario, "request.input"),
+                    templatesByFilename
             );
 
-            final Headers authorizationHeaders = getAuthorizationHeaders(scenario);
             final String requestUri = MapValueExtractor.extract(scenario, "request.uri");
-            final int expectedStatus = MapValueExtractor.extractOrDefault(scenario, "expectation.status", 200);
-
+            int expectedStatus;
+            if (scenarioFeature != null && launchDarklyFeature == false) {
+                expectedStatus = MapValueExtractor.extractOrDefault(scenario, "expectation.status", 200, launchDarklyFeature);
+            } else {
+                expectedStatus = MapValueExtractor.extractOrDefault(scenario, "expectation.status", 200);
+            }
             String actualResponseBody =
-                SerenityRest
-                    .given()
-                    .headers(authorizationHeaders)
-                    .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE)
-                    .body(requestBody)
-                    .when()
-                    .post(requestUri)
-                    .then()
-                    .statusCode(expectedStatus)
-                    .and()
-                    .extract()
-                    .body()
-                    .asString();
+                    SerenityRest
+                            .given()
+                            .headers(authorizationHeaders)
+                            .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE)
+                            .body(requestBody)
+                            .when()
+                            .post(requestUri)
+                            .then()
+                            .log().ifError()
+                            .log().ifValidationFails()
+                            .statusCode(expectedStatus)
+                            .and()
+                            .extract()
+                            .body()
+                            .asString();
+
+            System.out.println("ACTUAL RESPONSE: \n" + actualResponseBody);
```

Comment:
@burale we should probably look to remove this output now.

Reply:
Removed